actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -136,17 +136,27 @@ def header(routes)
         @buffer << draw_header(routes)
       end
 
-      def no_routes
-        @buffer << <<-MESSAGE.strip_heredoc
+      def no_routes(routes, filter)
+        @buffer << case true
+        when routes.none? then <<-MESSAGE.strip_heredoc
           You don't have any routes defined!
 
           Please add some routes in config/routes.rb.
-
-          For more information about routes, see the Rails guide: http://guides.rubyonrails.org/routing.html.
           MESSAGE
+        when !controller_exists?(filter)
+          "The controller #{filter} does not exist!"
+        else
+          "No routes were found for this controller"
+        end
+        @buffer << "For more information about routes, see the Rails guide: http://guides.rubyonrails.org/routing.html."
       end
 
       private
+        def controller_exists?(filter)
+          filter = filter.camelize << 'Controller' if !filter.end_with?('Controller')
+          !!filter.safe_constantize
```

Comment:
You're negating the return value when you call this method, so why not bake it in and do: 

```ruby
def missing_controller?(controller_name)
  [ controller_name.camelize, "#{controller_name.camelize}Controller" ].none?(&:safe_constantize)
end
```